src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -108,49 +77,39 @@ internal ReturnTypeAndDiagnostics(TypeSymbol returnType, ImmutableArray<Diagnost
         /// </summary>
         internal Binder ScopeBinder { get; }
 
+        public Binder ParameterBinder => _binder;
+
         internal void GetDeclarationDiagnostics(DiagnosticBag addTo)
         {
-            // Force attribute binding for diagnostics
+            // Force complete type parameters
             foreach (var typeParam in _typeParameters)
             {
-                typeParam.GetAttributesBag(addTo);
+                typeParam.ForceComplete(null, default(CancellationToken));
             }
 
             // force lazy init
             ComputeParameters();
 
-            foreach (var p in _syntax.ParameterList.Parameters)
+            foreach (var p in _lazyParameters)
             {
-                if (p.IsArgList)
-                {
-                    addTo.Add(ErrorCode.ERR_IllegalVarArgs, p.Location);
-                }
+                // Force complete parameters to retrieve all diagnostics
+                p.ForceComplete(null, default(CancellationToken));
             }
 
             ComputeReturnType();
 
-            var diags = ImmutableInterlocked.InterlockedExchange(ref _diagnostics, default(ImmutableArray<Diagnostic>));
-            if (!diags.IsDefault)
-            {
-                addTo.AddRange(diags);
-                addTo.AddRange(_lazyParametersAndDiagnostics.Diagnostics);
-                // Note _lazyParametersAndDiagnostics and _lazyReturnTypeAndDiagnostics
-                // are computed always, but _lazyTypeParameterConstraintsAndDiagnostics
-                // is only computed if there are constraints.
-                if (_lazyTypeParameterConstraintsAndDiagnostics != null)
-                {
-                    addTo.AddRange(_lazyTypeParameterConstraintsAndDiagnostics.Diagnostics);
-                }
-                addTo.AddRange(_lazyReturnTypeAndDiagnostics.Diagnostics);
-            }
+            addTo.AddRange(_declarationDiagnostics);
         }
 
+        internal override void AddDeclarationDiagnostics(DiagnosticBag diagnostics)
+            => _declarationDiagnostics.AddRange(diagnostics);
```

Comment:
>_declarationDiagnostics.AddRange(diagnostics) [](start = 15, length = 45)

Would there be an advantage of checking diagnostics.IsEmptyWithoutResolution before calling AddRange?